### PR TITLE
Allow pragma(inline, knownAtCTFE); version 2

### PR DESCRIPTION
**Pull request by BorisCarvajal**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
If buildkite libmir/mir still fails please force push to restart CI

**Comment by PetarKirov** [comment]
@BorisCarvajal You should have started your commit message with "fix" so that the issue would get auto-closed when the PR is merged:

```
Fix issue 21378 - `pragma(inline)` should allow not only a bool literal but any CTFE expression that converts to bool
```

**Comment by BorisCarvajal** [comment]
@PetarKirov sorry man I forgot it again

**Comment by PetarKirov** [comment]
No worries, keep up the excellent work!

**Comment by Geod24** [comment]
More important than auto-closing, this won't get mentioned in the changelog.

File: test/fail_compilation/pragmas.d [hunk 1]
```diff
@@ -24,7 +23,7 @@ void test2()
 
 void test3()
 {
-    pragma(inline, "string");
+    pragma(inline, "string"); // works now
 }
```
Comment (Geod24) [inline test/fail_compilation/pragmas.d:27]: Should have just removed the line IMO

**Comment by BorisCarvajal** [comment]
> More important than auto-closing, this won't get mentioned in the changelog.

#11988

**Pull request merged**